akka-http/src/main/scala/akka/http/javadsl/server/HttpApp.scala

```diff
@@ -16,7 +16,7 @@ import akka.http.scaladsl.Http.ServerBinding
 abstract class HttpApp
   extends AllDirectives
   with HttpServiceBase {
-  protected def createRoute(): Route
+  protected[javadsl] def createRoute(): Route
```

Comment:
It's needed for `testAppRoute` above. Maybe we should make it public in the first place.